cni/pkg/ambient/informers.go

```diff
@@ -133,18 +133,25 @@ func (s *Server) Reconcile(input any) error {
 	event := input.(controllers.Event)
 	log := log.WithLabels("type", event.Event)
 	pod := event.Latest().(*corev1.Pod)
+	log.Infof("Pod event for %s/%s(%s)", pod.Namespace, pod.Name, pod.Status.PodIP)
 	if ztunnelPod(pod) {
 		return s.ReconcileZtunnel()
 	}
 	switch event.Event {
 	case controllers.EventAdd:
+		if s.redirectMode == EbpfMode {
+			newPod := event.New.(*corev1.Pod)
+			if s.shouldPodBeInAmbient(newPod) {
+				s.AddPodToMesh(pod, "")
+			}
+		}
 	case controllers.EventUpdate:
 		// For update, we just need to handle opt outs
 		newPod := event.New.(*corev1.Pod)
 		oldPod := event.Old.(*corev1.Pod)
 		if ambientpod.PodHasOptOut(newPod) && !ambientpod.PodHasOptOut(oldPod) {
 			log.Debugf("Pod %s matches opt out, but was not before, removing from mesh", newPod.Name)
-			DelPodFromMesh(newPod)
+			s.DelPodFromMesh(newPod)
```

Comment:
Same here, DelPodFromMesh is not a receiver of the Server struct. Curious why this was changed?